sky/serve/replica_managers.py

```diff
@@ -327,15 +386,21 @@ def to_replica_status(self) -> serve_state.ReplicaStatus:
 class ReplicaInfo:
     """Replica info for each replica."""
 
+    _VERSION = 0
+
     def __init__(self, replica_id: int, cluster_name: str, replica_port: str,
+                 is_spot: bool, location: Optional[spot_policies.Location],
                  version: int) -> None:
+        self._version = self._VERSION
         self.replica_id: int = replica_id
         self.cluster_name: str = cluster_name
         self.version: int = version
         self.replica_port: str = replica_port
         self.first_not_ready_time: Optional[float] = None
         self.consecutive_failure_times: List[float] = []
         self.status_property: ReplicaStatusProperty = ReplicaStatusProperty()
+        self.is_spot: bool = is_spot
+        self.location: Optional[spot_policies.Location] = location
```

Comment:
should we just use the `handle.launched_resources` as part of the `ReplicaInfo`?

Reply:
I think the current version seems fine. Maybe in the future when we need to track the full resources we can store it.